### PR TITLE
modified interceptor to look for decrypted documents in payload and a…

### PR DESCRIPTION
**Pull request by rpinna**
…lways try to decrypt if they are present

### CODE REVIEW

File: src/app/services/iron/iron-http-interceptor.ts [hunk 1]
```diff
@@ -65,24 +64,40 @@ export class IronHttpInterceptor implements HttpInterceptor {
         );
     }
 
+    private isDecryptable(response: HttpResponse<any>): boolean {
```
Comment (madison-kerndt) [inline src/app/services/iron/iron-http-interceptor.ts]: You should be able to write this as: 

```javascript
    private isDecryptable(response: HttpResponse<any>): boolean {
        return response.body instanceof Array ? response.body.every((item) => EncryptedDocument.is(item)) : EncryptedDocument.is(response.body);
    }
```

just to clean up the code a little bit.

File: package.json [hunk 2]
```diff
@@ -53,4 +53,4 @@
         "jsxBracketSameLine": true,
         "arrowParens": "always"
     }
-}
\ No newline at end of file
+}
```
[the remark this answers is not in the document]
Comment (rpinna) [reply package.json]: agreed, i'll make sure i adopt that practice.

File: src/app/services/iron/iron-http-interceptor.ts [hunk 3]
```diff
@@ -49,14 +49,13 @@ export class IronHttpInterceptor implements HttpInterceptor {
...
 
-        // if policy, decrypt the response on the way back in
         if (policy.decrypt) {
             return next.handle(req)
                 .pipe(
                     tap((evt) => this.trace('pre-decrypt', policy, req, evt)),
                     switchMap((evt) => this.decrypt(evt)),
                     tap((devt) => this.trace('post-decrypt', devt))
-                );
+                )
```
[the remark this answers is not in the document]
Comment (rpinna) [reply src/app/services/iron/iron-http-interceptor.ts]: My bad.

**Comment by rpinna** [comment]
Made changes per comments.

File: src/app/services/iron/iron-http-interceptor.ts [hunk 4]
```diff
@@ -65,24 +64,40 @@ export class IronHttpInterceptor implements HttpInterceptor {
...
         const response = evt as HttpResponse<any>;
-        if (!response || !response.body) {
+        if (!this.isDecryptable(response)) {
+            this.trace('not decryptable')
             return of(evt);
         }
 
         // we may need to decrypt a list of encrypted documents (GET /)
         if (response.body instanceof Array) {
+            this.trace('decrypting a list');
```
[the remark this answers is not in the document]
Comment (rpinna) [reply src/app/services/iron/iron-http-interceptor.ts]: I set up the interceptor to take a user defined trace function

**Pull request merged**